Spanner/src/Operation.php

```diff
@@ -224,7 +230,13 @@ public function execute(Session $session, $sql, array $options = [])
 
         $context = $this->pluck('transactionContext', $options);
 
-        $call = function ($resumeToken = null) use ($session, $sql, $options) {
+        // Initially with begin, transactionId will be null.
+        // Once generated it will get populated from Result->rows()
+        // Incase of stream failure, the transactionId is preserved here to reuse.
+        $call = function ($resumeToken = null) use ($session, $sql, $options, &$transactionId) {
+            if ( !is_null($transactionId) and !is_array($transactionId) ) {
+                $options['transaction'] = ['id' => $transactionId];
```

Comment:
Would you want to handle the case when user sends an `$options` array like this?
```[
    'transaction' => ['id' => 'inline']
]```

Reply:
For the ILB to work, the user needs to call Database class functions with options ['begin' => true]. Refer `Database->execute` parameters here: https://github.com/googleapis/google-cloud-php/blob/076287b03e801bfaff4f43809de8c4bfc7c9e5d6/Spanner/src/Database.php#L1599